crates/turborepo-lib/src/shim.rs

```diff
@@ -281,6 +278,18 @@ impl RepoState {
     }
 }
 
+/// If the local turbo path doesn't exist or if we are local turbo, then we go
+/// ahead and run the Go code linked in the current binary.
+fn should_run_current_turbo(local_turbo_path: &Path) -> Result<bool> {
+    // Note we must check if local_turbo_path exists before we
+    // canonicalize the path, otherwise we'll get an error.
+    if !local_turbo_path.exists() {
+        return Ok(true);
+    }
+
+    Ok(local_turbo_path.canonicalize()? == current_exe()?.canonicalize()?)
```

Comment:
Can we avoid hitting the filesystem twice here?
```suggestion
    match local_turbo_path.canonicalize() {
        Ok(local_turbo_path) => Ok(local_turbo_path == current_exe()?.canonicalize()?),
        _ => Ok(true),
    }
```